uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Helper/ContactActionsMenu/WorkflowActionGroup.php

```diff
@@ -154,16 +138,12 @@ private function getWorkflowNewTaskButton() {
    *
    * @return ActionsGroupButtonItem
    */
-  private function getWorkflowNewDocumentButton() {
-    $url = CRM_Utils_System::url(
-      'civicrm/tasksassignments/dashboard#/documents',
-      "reset=1&cid=$this->contactID&openModal=document"
-    );
+  private function getNewDocumentButton() {
     $params = [
       'label' => 'New Document',
       'class' => 'btn btn-primary-outline',
       'icon' => 'fa fa-id-card-o',
-      'url' => $url
+      'url' => $this->getButtonUrl(['openModal' => 'document'], 'documents')
```

Comment:
What do you think of going a bit further and avoid having the second parameter? We can do that if we create separate methods for each tab. Example:

```php
private function getDocumentsTabUrl($queryParameters) {
  return $this->getButtonUrl($queryParameters, 'documents');
}

private function getTasksTabUrl($queryParameters) {
  return $this->getButtonUrl($queryParameters, 'tasks');
}
```

Then, instead of something like:

```php
$this->getButtonUrl(['openModal' => 'document'], 'documents')
```

You could have this:

```php
$this->getDocumentsTabUrl(['openModal' => 'document']);
```

Also, while writing this, I realized that the `getButtonUrl()` name might be a bit confusing. For me, it gives the impression that it will return a URL to a button when it's actually returning a URL to one of the tabs in the T&A dashboard. I think we should update that name to something more descriptive